pkg/services/internal/whatsonchain/service.go

```diff
@@ -39,10 +42,12 @@ func New(httpClient *resty.Client, logger *slog.Logger, apiKey string, network d
 		SetRetryMaxWaitTime(Retries * RetriesWaitTime)
 
 	return &WhatsOnChain{
-		httpClient: client,
-		apiKey:     apiKey,
-		url:        fmt.Sprintf("https://api.whatsonchain.com/v1/bsv/%s", network),
-		logger:     logger,
+		httpClient:        client,
+		apiKey:            config.ApiKey,
+		url:               fmt.Sprintf("https://api.whatsonchain.com/v1/bsv/%s", network),
+		logger:            logger,
+		bsvExchangeRate:   config.BsvExchangeRate,
+		bsvUpdateInterval: config.BsvUpdateInterval,
```

Comment:
set it with default so it doesn't need to be set every time we Update the exchange rate